### PR TITLE
Homeroom page additions

### PR DESCRIPTION
**Pull request by thfield**
This adds student attendance and office referral count for the current school year to the homeroom page.  It doesn't look like this was a requested feature, but I thought it could be useful for the "at-a-glance"  homeroom page.

screen shot 2015-05-30 at 2 07 24 pm

### CODE REVIEW

**Comment by thfield** [comment]
I did not, however, write any specs for these additions.  So, there's that.

**Comment by alexsoble** [comment]
@thfield — Wow, very cool! We are meeting with our Somerville teacher partners tomorrow and will ask them for their input. Thank you Tyler!

**Comment by thfield** [comment]
would love to hear if this is useful to teachers.  Just added feature to change highlighting bubbles on the fly.  The js code could be adapted easily to other table values if this is useful.

example

**Comment by thfield** [comment]
UI adjustments:  move spinner above column, add tooltip
example

**Comment by alexsoble** [comment]
Hi @thfield! We got feedback from the Somerville teacher team about the first screenshot you posted, adding student attendance and office referral count to the roster view: 

> "This would definitely be a useful tool. Since we would be able to hide it from view and pull it up with just a single click, it would be very helpful to us."

For the 2nd & 3rd screenshots (added feature of changing highlighting bubbles on the fly) I'm concerned that the new feature adds complexity. We haven't heard the Somerville teachers ask for the ability to customize the highlighting bubbles. 

I'd like to steer clear of building new features unless we have a direct ask from an end user in Somerville. That will help us keep the core product as lean and uncomplicated as possible.

**Comment by thfield** [comment]
@alexsoble totally understand not wanting to include features that aren't asked for.  Changing highlighting bubbles on the fly was something I thought would be cool, and it turns out to be pretty easy to implement.  If that feature request ever comes along, keep this in mind... 
I'll comment out the on-the-fly lines of code for now.

**Comment by thfield** [comment]
hey @alexsoble since you reassured me my additions were NOT overly annoying...  I made my little spinner additions hidden by default, and toggleable with a click.  That might be useful for other features that are a little more "advanced" for typical use but still could prove helpful.
04d9fcb61bc24d941c5318f9ee47b3ec

**Comment by alexsoble** [comment]
Hi @thfield — still like the idea of adding attendance and behavior counts to the roster. The "Toggle Advanced Controls" and other extra JavaScript adds more complexity though and we haven't heard any teachers or administrators ask for it. If you could strip down this PR to just include the attendance & behavior counts on the roster, that would be perfect. Let's also cut out the warning bubbles on attendance/behavior now since we don't have a clear sense from our Somerville partners what those warning levels should look like. Thank you! This will be a great addition!

**Comment by thfield** [comment]
Hey @alexsoble just pushed a version that strips away extra chrome/complexity, looks like there are conflicts, maybe because you've made additions since I worked on this? I didn't comb through the rails code to remove extra fluff though, if attendance warning levels eventually make sense it would be easy to re-implement.

**Comment by alexsoble** [comment]
@thfield Awesome, if you could checkout changes to all files except the ones that affect the feature (`_student_row.html.erb`, `index.html.erb`, `student_presenter.rb` ... ) and then squash your commits we should be good to merge!

**Comment by thfield** [comment]
hey @alexsoble I did some rebaseing/squashing, but while looking into why the build fails I realized I should refactor my code to be more aligned with the mcas/star result presenters.  I'll have some time tomorrow to work on this

**Comment by thfield** [comment]
@alexsoble finally got around to making those changes!  travis-ci checks are still running as I'm writing this, but all the specs cleared on my local machine and i merged conflicts from current master branch

File: spec/features/roster_feature_spec.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
         expect(student_rows.size).to eq 1
       end
     end
-    context 'one student in homeroom' do
+    context 'three students in homeroom' do
```
Comment (alexsoble) [inline spec/features/roster_feature_spec.rb]: Thanks for catching this!

**Comment by alexsoble** [comment]
@thfield Woohoo! I don't think we have a definitive point of view from Somerville Public Schools yet on what a warning level on attendance/tardies/behavior might be, but the rest looks great! Merging now.

**Pull request merged**